### ISSUE TITLE
Lemminx extension not running diagnostics participant

### ISSUE THREAD

**Issue by rzgry** [eclipse/lemminx#945]
My lemminx extension suddenly seems to not be running the diagnostic participant.  Strange thing is that it is working fine for @kathrynkodama.

I added some log messages in the language feature participants https://github.com/OpenLiberty/liberty-language-server/commit/2b1135cf4a717c90428ba4aa33eab7c12d18ffd1 but dont see any output from the diagnostic participant or any lsp messages for diagnostics in the trace, but the output for hover / completion is working fine.

I have validation enabled in my vscode xml settings.

![image](https://user-images.githubusercontent.com/16768710/101674828-fcd64080-3a26-11eb-9547-60fe103b695b.png)

Any thoughts on why it would not be running diagnostics participant on my machine? 

Trace:
```
[Trace - 1:57:27 p.m.] Sending request 'initialize - (0)'.
[Trace - 1:57:28 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:28 p.m.] Dec. 09, 2020 01:57:28 org.eclipse.lemminx.XMLLanguageServer initialize()
Message: Initializing XML Language server
LemMinX Server info:
 - Version : 0.14.0-SNAPSHOT
 - Java : /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
 - Git 3ea8abb - Improve error range for ETagUnterminated (#877)
[Trace - 1:57:28 p.m.] Received response 'initialize - (0)' in 647ms.
[Trace - 1:57:28 p.m.] Sending notification 'initialized'.
[Trace - 1:57:28 p.m.] Sending notification 'workspace/didChangeConfiguration'.
[Trace - 1:57:28 p.m.] Sending notification 'textDocument/didOpen'.
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (1)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (1)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (2)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (2)'. Processing request took 0ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (3)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (3)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Sending request 'textDocument/codeAction - (1)'.
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (4)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (4)'. Processing request took 5ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (5)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (5)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (6)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (6)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (7)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (7)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Sending request 'textDocument/documentLink - (2)'.
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (8)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (8)'. Processing request took 2ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (9)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (9)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (10)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (10)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (11)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (11)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (12)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (12)'. Processing request took 1ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (13)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (13)'. Processing request took 2ms
[Trace - 1:57:28 p.m.] Received request 'client/registerCapability - (14)'.
[Trace - 1:57:28 p.m.] Sending response 'client/registerCapability - (14)'. Processing request took 2ms
[Trace - 1:57:28 p.m.] Received response 'textDocument/codeAction - (1)' in 55ms.
[Trace - 1:57:28 p.m.] Sending request 'textDocument/documentSymbol - (3)'.
[Trace - 1:57:28 p.m.] Received response 'textDocument/documentSymbol - (3)' in 16ms.
[Trace - 1:57:28 p.m.] Received response 'textDocument/documentLink - (2)' in 147ms.
[Trace - 1:57:28 p.m.] Sending request 'textDocument/foldingRange - (4)'.
[Trace - 1:57:28 p.m.] Received response 'textDocument/foldingRange - (4)' in 5ms.
[Trace - 1:57:31 p.m.] Sending request 'textDocument/foldingRange - (5)'.
[Trace - 1:57:31 p.m.] Received response 'textDocument/foldingRange - (5)' in 2ms.
[Trace - 1:57:36 p.m.] Sending request 'textDocument/hover - (6)'.
[Trace - 1:57:36 p.m.] Sending notification '$/cancelRequest'.
[Trace - 1:57:36 p.m.] Received response 'textDocument/hover - (6)' in 46ms. Request failed: The request (id: 6, method: 'textDocument/hover') has been cancelled (-32800).
[Trace - 1:57:36 p.m.] Sending request 'textDocument/hover - (7)'.
[Trace - 1:57:36 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:36 p.m.] Dec. 09, 2020 01:57:36 io.openliberty.lemminx.liberty.LibertyHoverParticipant onText()
Message: Running Hover
[Trace - 1:57:37 p.m.] Received response 'textDocument/hover - (7)' in 903ms.
[Trace - 1:57:45 p.m.] Sending request 'textDocument/hover - (8)'.
[Trace - 1:57:45 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:45 p.m.] Dec. 09, 2020 01:57:45 io.openliberty.lemminx.liberty.LibertyHoverParticipant onText()
Message: Running Hover
[Trace - 1:57:45 p.m.] Received response 'textDocument/hover - (8)' in 3ms.
[Trace - 1:57:50 p.m.] Sending request 'textDocument/hover - (9)'.
[Trace - 1:57:50 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:50 p.m.] Dec. 09, 2020 01:57:50 io.openliberty.lemminx.liberty.LibertyHoverParticipant onText()
Message: Running Hover
[Trace - 1:57:50 p.m.] Received response 'textDocument/hover - (9)' in 3ms.
[Trace - 1:57:51 p.m.] Sending request 'textDocument/hover - (10)'.
[Trace - 1:57:51 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:51 p.m.] Dec. 09, 2020 01:57:51 io.openliberty.lemminx.liberty.LibertyHoverParticipant onText()
Message: Running Hover
[Trace - 1:57:51 p.m.] Received response 'textDocument/hover - (10)' in 2ms.
[Trace - 1:57:56 p.m.] Sending request 'textDocument/completion - (11)'.
[Trace - 1:57:56 p.m.] Received notification 'window/logMessage'.
[Info  - 1:57:56 p.m.] Dec. 09, 2020 01:57:56 io.openliberty.lemminx.liberty.LibertyCompletionParticipant onXMLContent()
Message: Running completion
[Trace - 1:57:56 p.m.] Received response 'textDocument/completion - (11)' in 34ms.
[Trace - 1:58:02 p.m.] Sending notification 'textDocument/didChange'.
[Trace - 1:58:02 p.m.] Sending request 'textDocument/foldingRange - (12)'.
[Trace - 1:58:02 p.m.] Received response 'textDocument/foldingRange - (12)' in 4ms.
[Trace - 1:58:02 p.m.] Sending request 'textDocument/codeAction - (13)'.
[Trace - 1:58:02 p.m.] Received response 'textDocument/codeAction - (13)' in 5ms.
[Trace - 1:58:02 p.m.] Sending request 'textDocument/documentSymbol - (14)'.
[Trace - 1:58:02 p.m.] Received response 'textDocument/documentSymbol - (14)' in 2ms.
[Trace - 1:58:03 p.m.] Sending request 'textDocument/documentLink - (15)'.
[Trace - 1:58:03 p.m.] Received response 'textDocument/documentLink - (15)' in 1ms.
[Trace - 1:58:03 p.m.] Sending notification 'textDocument/didSave'.

```

**Comment by angelozerr** [comment]
@rzgry I suppose you have none validation (even if your XML file doesn't respect the XSD schema). I think you have an exception in your participant. We should take care of with https://github.com/eclipse/lemminx/issues/946

**Comment by datho7561** [comment]
Looking into this...

**Comment by datho7561** [comment]
@rzgry Are you referencing the extension JAR with an absolute reference, and if not, could you please try that? I found that the extension doesn't load if I had `~` in the path that references the JAR.

**Comment by datho7561** [comment]
I haven't been able to reproduce the issue with the latest commit of lemminx-liberty. We merged PR #947 which may help. Please let me know if you can still reproduce the issue.

**Comment by rzgry** [comment]
Updated my vscode-xml to the latest development build from https://download.jboss.org/jbosstools/vscode-xml/staging/?C=M;O=D and diagnostics are being displayed again.
